docs/mobile/exercises/README.md

```diff
@@ -0,0 +1,294 @@
+# Mobile exercises
+
+Writing a DataCamp for Mobile is quite different than writing a DataCamp
+[desktop course](../../courses/readme.md). The upside is that you can reach a
+much broader audience because the convenience is higher and the barrier to entry
+is lower. The challenge is that there are many more constraints to work
+within. Primary among these are the restricted screen real estate and the lack
+of an interactive interpreter. These constraints require you to think carefully
+about the way you introduce concepts and the examples that you use.
+
+# Basic exercise structure
+
+There are roughly 5 exercise types available to you as an author on DataCamp
+Mobile, but they all share a similar structure. In general (with a few
+exceptions), you can think of an exercise as a short code vignette comprising
+various blocks in the following structure:
+
+```yaml
+# print()
+- key: c11dbdf1-49c9-408a-a1e4-885431af2196
+  context: "`print()` outputs to the screen. Whatever you put between the parentheses will be output! E.g. `print(5)` will output `5`!"
+  question: "" # this is the default
+  code: "print(42)"
+  output: "42"
+  distractor_output:
+    - option: "4"
+      feedback: "`print()` will output all of whatever is between the parentheses, not just the first part."
+    - option: "2"
+      feedback: "`print()` will output all of whatever is between the parentheses, not just the last part."
+```
+
+Let's dig into each of the blocks...
+
+## topic
+
+```yaml
+# print()
+```
+
+Even though it's not part of the exercise itself, it's a very good idea to
+comment your exercises so you can quickly remind yourself what each exercise is
+designed to teach or test. In fact, it can be helpful to start by outlining a
+whole lesson with a scaffolding of comments before expanding the exercises.
+
+## key
+
+```yaml
+- key: c11dbdf1-49c9-408a-a1e4-885431af2196
+```
+
+This is a unique UUID for the exercise.
+
+## context
+
+```yaml
+  context: "`print()` outputs to the screen. Whatever you put between the parentheses will be output! E.g. `print(5)` will output `5`!"
+```
+
+This is where you can provide exposition necessary to introduce a new
+concept. It should be kept as minimal as possible (no longer than 150
+characters). It's good to shoot for a lighter, more informal feel here than on
+desktop. This field supports markdown elements like `inline code` and
+**emboldening**. Emojis are not just supported, they're encouraged!
+
+## question
+
+```yaml
+  question: "" # this is the default
+```
+
+This field is always optional. It will override the default question for the
```

Comment:
Why is the `question` field discussed in the same section as the compulsory fields, even though it is optional? I think this belongs later on.

Reply:
Even though it's not compulsory in the backend, it always renders in the front-end (there are fallback defaults), so its helpful for establishing the gestalt of what an exercise looks/feels like.